src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -130,14 +317,19 @@ const _calculateDistancesAlongRoute = (waypointModelList, airport) => {
  */
 const _preSpawn = (spawnPatternJson, airport) => {
     // distance between each arriving aircraft, in nm
-    const entrailDistance = spawnPatternJson.speed / spawnPatternJson.rate;
+    const airspaceCeiling = airport.maxAssignableAltitude;
+    const spawnSpeed = spawnPatternJson.speed;
+    const spawnAltitude = spawnPatternJson.altitude;
+    const entrailDistance = spawnSpeed / spawnPatternJson.rate;
     const routeModel = new RouteModel(spawnPatternJson.route);
     const waypointModelList = routeModel.waypoints;
-    const { totalDistance } = _calculateDistancesAlongRoute(waypointModelList, airport);
+    const totalDistance = _calculateDistancesAlongRoute(waypointModelList, airport);
     // calculate number of offsets
     const spawnOffsets = _assembleSpawnOffsets(entrailDistance, totalDistance);
     // calculate heading, nextFix and position data to be used when creating an `AircraftModel` along a route
-    const spawnPositions = _calculateSpawnPositions(waypointModelList, spawnOffsets, airport);
+    const spawnPositions = _calculateSpawnPositionsAndAltitudes(
+        waypointModelList, spawnOffsets, spawnSpeed, spawnAltitude, totalDistance, airspaceCeiling
```

Comment:
these should each be on their own line
```suggestion
        waypointModelList, 
        spawnOffsets, 
        spawnSpeed, 
        spawnAltitude, 
        totalDistance, 
        airspaceCeiling
```

Reply:
done